decoder.go

```diff
@@ -0,0 +1,133 @@
+package drum
+
+import (
+	"encoding/binary"
+	"errors"
+	"io"
+	"log"
+	"os"
+	"strings"
+)
+
+// DecodeFile decodes the drum machine file found at the provided path
+// and returns a pointer to a parsed pattern which is the entry point to the
+// rest of the data.
+func DecodeFile(path string) (*Pattern, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+
+	p := &Pattern{}
+	if err := NewDecoder(file).Decode(p); err != nil {
+		log.Fatal(err)
```

Comment:
Since the function returns an error, is it appropriate to also have exiting the entire program as a consequence?